src/parseTools.js

```diff
@@ -1096,17 +1106,14 @@ function receiveI64ParamAsI32s(name) {
   return '';
 }
 
-// TODO: use this in library_wasi.js and other places. but we need to add an
-//       error-handling hook here.
-function receiveI64ParamAsDouble(name) {
+function receiveI64ParamAsDouble(name, onError) {
```

Comment:
End users may also have been using `receiveI64ParamAsDouble` as they may need to convert I64 pairs to doubles as well? This change will break their code?

Reply:
I guess making the `onEerror` parameter optional and adding a test for that case would be address that concern?

Reply:
yeah, certainly, that would make the old form work as well.

Reply:
I ended up going with a completely new function `receiveI64ParamAsI53` rather and deprecating the old/unused `receiveI64ParamAsDouble` .    This avoids over-complicating the new code with unused/legacy behaviour and gives the functions names more semantic meaning (easier to grep for usage, for example).